src/kernels/jupyter/interpreter/jupyterInterpreterOldCacheStateStore.node.ts

```diff
@@ -26,15 +27,15 @@ export class JupyterInterpreterOldCacheStateStore {
     ) {
         // Cache stores to keep track of jupyter interpreters found.
         const workspaceState =
-            persistentStateFactory.createWorkspacePersistentState<string>('DS-VSC-JupyterInterpreter');
-        const globalState = persistentStateFactory.createGlobalPersistentState<string>('DS-VSC-JupyterInterpreter');
+            persistentStateFactory.createWorkspacePersistentState<Uri>('DS-VSC-JupyterInterpreter-2');
+        const globalState = persistentStateFactory.createGlobalPersistentState<Uri>('DS-VSC-JupyterInterpreter-2');
         this.workspaceJupyterInterpreter = { state: workspaceState };
         this.globalJupyterInterpreter = { state: globalState };
     }
     private get cacheStore(): CacheInfo {
         return this.workspace.hasWorkspaceFolders ? this.workspaceJupyterInterpreter : this.globalJupyterInterpreter;
     }
-    public getCachedInterpreterPath(): string | undefined {
+    public getCachedInterpreterPath(): Uri | undefined {
         return this.cacheStore.state.value;
```

Comment:
I don't think we can cache (serlalize Uris) in this manner. Basically it would only get serialized as a JSON, hence all of the methods would not be availabel upon deserializing and things could fall over. I think we should store the string value and use `Uri.file` when reading back from state.value

Reply:
Yes the JSON only works if something had actually called 'fsPath' before. I'm actually working on this part right now.